### DISCUSSION TITLE
Unexpected output from GFM tables (mdast-util-gfm-table)

### DISCUSSION THREAD

**Discussion by charliematters** [syntax-tree/unist#57] [Q&A] [answered]
Hi,

I hope this is the right place to ask about `mdast-util-gfm-table`, but I ended up here from [this issues page](https://github.com/syntax-tree/mdast-util-gfm-table/issues/new/choose)

## Context
We've recently migrated from react-markdown 4 to 7, and I'm running into a issue where the existing markdown data is producing different results.

We want to use the GFM tables extension, but we do NOT want the autolinking functionality in GFM as we are recording links that we don't want people visiting - think "www.badguys.com sucks"

Previously we used the whole remark-gfm package, and then went used [remark-disable-tokenizers](https://github.com/zestedesavoir/zmarkdown/tree/remark-disable-tokenizers%401.1.0/packages/remark-disable-tokenizers#caveats) to remove the autolinking. This worked because the autolinks came out as `linkReference`s and everything else came out as `link`s. 

In newer versions of remark, this distinction has been dropped , so we had to stop using remark-gfm (as it is an all-or-nothing affair) and just use mdast-util-gfm-table to provide us the tables.

## Problem
With new versions of remark, a heading that immediately follows a table gets added to the table.

With a slight modification to the example found [here](https://github.com/syntax-tree/mdast-util-gfm-table), I have this:
```javascript
import { fromMarkdown } from 'mdast-util-from-markdown'
import { gfmTableFromMarkdown, gfmTableToMarkdown } from 'mdast-util-gfm-table'
import { toMarkdown } from 'mdast-util-to-markdown'
import { gfmTable } from 'micromark-extension-gfm-table'

const tableMarkdown = `
| AA  | BB  | CC  |
|-----|-----|-----|
|  aa | bb  | cc  |
# Not included in table
`

const tree = fromMarkdown(tableMarkdown, {
  extensions: [gfmTable],
  mdastExtensions: [gfmTableFromMarkdown],
})

console.log(toMarkdown(tree, { extensions: [gfmTableToMarkdown()] }))
```

Which unfortunately produces this:
``` 
| AA                      | BB | CC |
| ----------------------- | -- | -- |
| aa                      | bb | cc |
| # Not included in table |    |    |
```

If you change the heading to a blockquote, we end up with what I'd expect:
```
| AA | BB | CC |
| -- | -- | -- |
| aa | bb | cc |

> Not included in table
```

I've been trying to work out whether this is just a vaguery in the GFM flavour, so I found the spec [here](https://github.github.com/gfm/#example-201) which says:
> The table is broken at the first empty line, or beginning of another block-level structure:

I wanted to check headings are considered block-level items, and [this site](https://daringfireball.net/projects/markdown/syntax#block) seems to agree by including both headings and blockquotes under the "Block Elements" section.

Finally, confident that this is a bug, I checked with a couple of online editors, all of which separate out the heading from the table. 
* [https://stackedit.io](https://stackedit.io/)
* [https://jbt.github.io/markdown-editor](https://jbt.github.io/markdown-editor)
* This page:

| AA  | BB  | CC  |
|-----|-----|-----|
|  aa | bb  | cc  |
# Not included in table

I suppose I could look into which libraries they are using, but I don't want to remove all the `react-markdown` dependencies we have just for some titles

**Comment by wooorm** [comment, marked as answer]
seems to be a bug!

**Reply by wooorm** [reply]
Thanks for the information. I need a clear head to tackle this but it shouldn’t be extremely time consuming. I’ll try tomorrow!

**Reply by charliematters** [reply]
Thanks very much!

**Reply by wooorm** [reply]
Moved it here: https://github.com/micromark/micromark-extension-gfm-table/issues/4
